spec/core/ics-026-routing-module/README.md

```diff
@@ -91,19 +91,19 @@ function onChanCloseConfirm(
     // defined by the module
 }
 
-function onRecvPacket(packet: Packet): bytes {
+function onRecvPacket(packet: Packet, relayer: string): bytes {
```

Comment:
Where the `relayer` comes from? Do we need to add this field to IBC messages?If yes, are there some properties we expect to hold, i.e., is it a problem if a relayer puts a string that is different from  it's address?

Reply:
Yes we would have to modify `RecvPacketMsg` to allow relayers to include their source address. If a relayer inputs an invalid address, the forward fee will simply be refunded.

It is incumbent on forward relayer to correctly provide address. If this does not happen, then packet processing will still continue as we do not allow relayers to prevent packet processing

Reply:
This relayer is just the message signer (destination address of relayer). The fee middleware holds a mapping from the destination address to relayer source address which it can put in acknowledgement. If they put wrong source address, they don't get paid. Packet processing is not affected